src/libmusig2.c

```diff
@@ -177,12 +177,21 @@ void musig2_context_free(musig2_context *mc) {
     }
 }
 
+/*** Free memory allocated in MuSig2 context aggr_R_list ***/
+void musig2_context_aggr_R_free(musig2_context *mc, int nr_messages) {
+    for (int l = 0; l < V * nr_messages ; l++) {
+        free(mc->aggr_R_list[l]);
+    }
+    free(mc->aggr_R_list);
+}
```

Comment:
This should be part of `musig2_context_free`. Also, I'm thinking, now that we include the batched `aggr_R_list` in `musig2_context`, then I think it also makes sense to include in this structure the number of messages.